Source/SwiftLintFramework/Reporters/SonarQubeReporter.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import SourceKittenFramework
+
+public struct SonarQubeReporter: Reporter {
+    public static let identifier = "sonarqube"
+    public static let isRealtime = false
+
+    public var description: String {
+        return "Reports violations in SonarQube import format."
+    }
+
+    public static func generateReport(_ violations: [StyleViolation]) -> String {
+        return toJSON(["issues": violations
+            .filter { $0.location.relativeFile != nil && $0.location.line != nil }
```

Comment:
Is there a reason to filter? We already have to use optional chaining below when creating the dictionary

Reply:
Exactly. It was kind of error handling, but I tried again without filtering, and I found it works. so I removed this filtering.